docs/core/geometry.md

```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
 
 ```{tip}
-The coordinate systems define - in an ideal setup - also the readout measurement directions. In such a case, a track prediction from the propagation, e.g. will already be in the correct frame of the measurement and residual or compatibility checks will not need additional coordinate transformations.
+The coordinate systems define - in an ideal setup - also the readout
+measurement directions. In such a case, a track prediction from the
+propagation, e.g. will already be in the correct frame of the measurement and
+residual or compatibility checks will not need additional coordinate
+transformations.
 ```
 
+### Plane surface
+
+![PlaneBounds](/figures/geometry/PlaneBounds.png)
+
+:::{doxygenclass} Acts::PlaneSurface
+---
+members: globalToLocal,localToGlobal,intersect,normal
+---
+:::
+
+### Disc surface
+
+![DiscBounds](/figures/geometry/DiscBounds.png)
+
+:::{doxygenclass} Acts::DiscSurface
+---
+members: globalToLocal,localToGlobal,intersect,normal
+---
+:::
+ 
+### Cylinder surface
+
+![CylinderBounds](/figures/geometry/CylinderBounds.png)
+
+:::{doxygenclass} Acts::CylinderSurface
+---
+members: globalToLocal,localToGlobal,intersect,normal
+---
+:::
+
+### Cone surface
+
+:::{doxygenclass} Acts::ConeSurface
+---
+members: globalToLocal,localToGlobal,intersect,normal
+---
+:::
+
+### Line surface
+
+{class}`Acts::LineSurface` is a special kind of surface that depends on a reference
+direction, typically the unit momentum direction of a particle. A point in
+space is considered *on surface* if and only if it coincides with the point of
+closest approach between the direction vector $\vec d$ and the line direction
+vector $\vec z$. As such, the function {func}`Acts::LineSurface::globalToLocal`
+can fail, if the argument position and direction do not fulfill this criterion.
+It if pure-virtual, meaning that it can not be instantiated on its own.
+
+:::{doxygenclass} Acts::LineSurface
+---
+members: globalToLocal,localToGlobal,intersect,normal
+---
+:::
+
+#### Straw surface
+
+:::{doxygenclass} Acts::StrawSurface
+---
+members: false
+---
+:::
+
+#### Perigee surface
+
+:::{doxygenclass} Acts::PerigeeSurface
+---
+members: false
+---
+:::
+
+
 ## Layer classes
 
-The `Layer` class is an extension of the `Surface` class that allows the
+
+The {class}`Acts::Layer` class is an extension of the {class}`Acts::Surface` class that allows the
 definition of sub surfaces (sensitive surfaces for modules, or extra material
 surfaces).
 
 The Layer can simply correspond to a 'virtual' surface in the detector
 description or represent a more complex object that may contain:
 
-* a representing surface, which is accessible via a `representingSurface()`
-* method an array of contained surfaces, accessible via `surfaceArray()` method
+* a representing surface, which is accessible via a {func}`Acts::Layer::surfaceRepresentation`
+* method an array of contained surfaces, accessible via {func}`Acts::Layer::surfaceArray` method
```

Comment:
I don't understand this bullet point (there is some grammatical problem I think)

Reply:
```suggestion
* an array of contained surfaces, accessible via {func}`Acts::Layer::surfaceArray` method
```